test/e2e/network/dns.go

```diff
@@ -654,6 +655,36 @@ var _ = common.SIGDescribe("DNS", func() {
 		}
 		validateDNSResults(ctx, f, pod, append(agnhostFileNames, jessieFileNames...))
 	})
+
+	framework.It("should work with a host that starts with a digit", framework.WithFeatureGate(features.RelaxedServiceNameValidation), func(ctx context.Context) {
+		svcName := "1kubernetes"
+		svc := v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: svcName,
+			},
+			Spec: v1.ServiceSpec{
+				Ports: []v1.ServicePort{{Port: 443}},
+			},
+		}
+
+		createServiceReportErr(ctx, f.ClientSet, f.Namespace.Name, &svc)
+		namesToResolve := []string{
+			fmt.Sprintf("%s.%s.svc.%s", svcName, f.Namespace.Name, framework.TestContext.ClusterDNSDomain),
+		}
+
+		// TODO: Validate both IPv4 and IPv6 families for dual-stack
```

Comment:
not really needed... there's no reason dual-stack in particular would break in this case